kolibri/core/assets/src/core-app/mediator.js

```diff
@@ -433,4 +455,58 @@ module.exports = class Mediator {
       url: messageMapUrl,
     };
   }
+  /**
+   * A method for registering content renderers for asynchronous loading and track
+   * which file types we have registered renderers for.
+   * @param  {String} kolibriModuleName name of the module.
+   * @param  {String[]} kolibriModuleUrls the URLs of the Javascript
+   * files that constitute the kolibriModule
+   * @param  {Object} contentTypes      Object of kind, array of extension mappings
+   */
+  registerContentRenderer(kolibriModuleName, kolibriModuleUrls, contentTypes) {
+    this._contentRendererUrls[kolibriModuleName] = kolibriModuleUrls;
+    Object.keys(contentTypes).forEach((kind) => {
+      if (!this._contentRendererRegistry[kind]) {
+        this._contentRendererRegistry[kind] = {};
+      }
+      contentTypes[kind].forEach((extension) => {
+        if (this._contentRendererRegistry[kind][extension]) {
+          logging.warn(`Two content renderers are registering for ${kind}/${extension}`);
+        } else {
+          this._contentRendererRegistry[kind][extension] = kolibriModuleName;
+        }
+      });
+    });
+  }
+  /**
+   * A method to retrieve a content renderer component.
+   * @param  {String} kind      content kind
+   * @param  {String} extension content extension
+   * @return {Promise}          Promise that resolves with loaded content renderer
```

Comment:
it's a little unclear to me what exactly "loaded content renderer" is. My best guess is a Vue component options object?